pkg/aws/sts/cache.go

```diff
@@ -95,7 +97,7 @@ func (c *credentialsCache) Expiring() chan *CachedCredentials {
 // CredentialsForRole looks for cached credentials or requests them from the STSGateway. Requested credentials
 // must have their ARN set.
 func (c *credentialsCache) CredentialsForRole(ctx context.Context, identity *RoleIdentity) (*Credentials, error) {
-	logger := log.WithFields(log.Fields{"pod.iam.role": identity.Role, "pod.iam.roleArn": identity.ARN})
+	logger := log.WithFields(log.Fields{"pod.iam.role": identity.Role, "pod.iam.roleArn": identity.Role.ARN})
```

Comment:
I think maybe this is worth pulling out into an `identity.LogFields()` func? That way we can expand to include the session name (and maybe) a sanitised external ID?

Reply:
Pulled out to function.